app/main.py

```diff
@@ -1,2 +1,65 @@
+from typing import Any
+
+
 class Dictionary:
-    pass
+    def __init__(self, initial_capacity: int = 8) -> None:
+        self.capacity = initial_capacity
+        self.size = 0
+        self.storage = [None] * self.capacity
+
+    class Node:
+        def __init__(self, key: Any, value: Any) -> None:
+            self.key = key
+            self.value = value
+            self.next = None
+
+    def _resize(self) -> None:
```

Comment:
When _resize() is called, it correctly rehashes and repositions nodes in the new new_storage array. However, if there are any collisions at the new index, the code will overwrite the existing chain by creating a new chain rather than properly appending it.

When adding nodes to new_storage, continue chaining nodes at the correct index in new_storage by appending to the end of the existing chain (rather than always assigning new_node.next = new_storage[new_index]).